jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -282,7 +279,35 @@ public PathResource(URI uri) throws IOException
      */
     public PathResource(URL url) throws IOException, URISyntaxException
     {
-        this(url.toURI());
+        this(toFile(url));
+    }
+
+    /**
+     * Converts a file {@link URL} to a {@link File}.
+     * 
+     * @param url
+     *            the {@link URL} to convert.
+     * @return the converted {@link File}.
+     */
+    private static File toFile(URL url)
+    {
+        checkFileURIScheme(url.getProtocol());
+        return new File(url.getPath());
```

Comment:
This is invalid.
url shouldn't be used like this.  (it will fail on Windows).
use the url.toURI() and only use the URI form, its the only form compatible with ALL OS's.

Reply:
Thank you for your return. 

Could you please tell me why you say that it will fail on windows ? (I may have used `URL#getFile` instead of `URL#getPath` if you prefer, but as far as I can see, both are equivalent and both work on windows).

As you can see in the issue I have created `URL#toURI` is not robust as it may fail if the URL contains "Invalid URI characters". this typically happen in an eclipse environment when you rely on the platform to retrieve URLs. As you can see, other people meet the same type of issue in eclipse : https://stackoverflow.com/questions/4494063/how-to-avoid-java-net-urisyntaxexception-in-url-touri/4494298